core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -420,7 +420,7 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
 
 	etx.BroadcastAt = &initialBroadcastAt
 
-	if sendError.IsNonceTooLowError() || sendError.IsReplacementUnderpriced() {
+	if sendError.IsNonceTooLowError() || sendError.IsTransactionAlreadyMined() || sendError.IsReplacementUnderpriced() {
```

Comment:
do we want to check for it here as well, since `IsNonceTooLowError` is being checked?

Reply:
Yes absolutely. It should be added in all places where IsNonceTooLowError is checked